packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts

```diff
@@ -660,8 +842,108 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
     }
   }
 
-  public async registerRevocationStatusList(): Promise<RegisterRevocationStatusListReturn> {
-    throw new AriesFrameworkError('Not implemented!')
+  public async registerRevocationStatusList(
+    agentContext: AgentContext,
+    { options, revocationStatusList }: IndyVdrRegisterRevocationStatusList
+  ): Promise<RegisterRevocationStatusListReturn> {
+    try {
+      // This will throw an error if trying to register a credential definition with a legacy indy identifier. We only support did:indy
+      // identifiers for registering, that will allow us to extract the namespace and means all stored records will use did:indy identifiers.
+      const { namespaceIdentifier, namespace } = parseIndyDid(revocationStatusList.issuerId)
+      const indyVdrPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService)
+      const pool = indyVdrPoolService.getPoolForNamespace(namespace)
+
+      agentContext.config.logger.debug(
+        `Registering revocation status list on ledger '${namespace}' with did '${revocationStatusList.issuerId}'`,
+        revocationStatusList
+      )
+
+      let writeRequest: CustomRequest
+      let didIndyRevocationStatusList: string
+
+      const { endorsedTransaction, endorserDid, endorserMode } = options
+      if (endorsedTransaction) {
+        agentContext.config.logger.debug(
+          `Preparing endorsed tx '${endorsedTransaction}' for submission on ledger '${namespace}' with did '${revocationStatusList.issuerId}'`,
+          revocationStatusList
+        )
+        writeRequest = new CustomRequest({ customRequest: endorsedTransaction })
+        // extract the seqNo from the endorsed transaction, which is contained in the ref field of the operation
+      } else {
+        const revocationRegistryDefinitionRequest = new RevocationRegistryEntryRequest({
+          submitterDid: namespaceIdentifier,
+          revocationRegistryEntry: {
+            ver: '1.0',
+            value: {
+              accum: revocationStatusList.currentAccumulator,
+            },
+          },
+          revocationRegistryDefinitionType: 'CL_ACCUM',
+          revocationRegistryDefinitionId: revocationStatusList.revRegDefId,
+        })
```

Comment:
We also need to pass the issued/revoked indices here. These are missing in the JS wrapper: https://github.com/hyperledger/indy-vdr/issues/165

Not sure if you can pass them with a ts-ignore for now and they will be passed on automatically to indy-vdr native lib or whether actual changes need to be published for this transaction can work: https://github.com/hyperledger/indy-vdr/issues/165

Reply:
It seems to be passing the entry value as a whole to indy-vdr, so I think it should just work by passing the values: https://github.com/hyperledger/indy-vdr/blob/main/wrappers/javascript/indy-vdr-nodejs/src/NodeJSIndyVdr.ts#L405

See: https://github.com/hyperledger/indy-node/blob/main/design/anoncreds.md#revoc_reg_entry-txn

Reply:
It is still a bit confusing, but the public API now also accepts the issued/revoked arrays as I assumed that that was the point. The changes between indy-sdk, indy-vdr, non-indy ledgers and anoncreds objects vs indy objects is causing quite some confusion ;).